flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogPartition.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents a partition object in catalog.
+ */
+public interface CatalogPartition {
+
+	/**
+	 * Get the PartitionSpec of the partition.
+	 *
+	 * @return the PartitionSpec of the partition.
+	 */
+	PartitionSpec getPartitionSpec();
```

Comment:
PartitionSpec is more like the name of a partition. Similar to the name of a table or function, which isn't included in the class definition, I think PartitionSpec should also stand out of Partition class for consistency. Partition class should care mostly about things other than naming, such as schema, stats, properties, etc.

Reply:
sounds good